src/protocols/secure_channel/PairingSession.h

```diff
@@ -162,44 +162,40 @@ class DLL_EXPORT PairingSession : public SessionDelegate
     CHIP_ERROR HandleStatusReport(System::PacketBufferHandle && msg, bool successExpected)
     {
         Protocols::SecureChannel::StatusReport report;
-        CHIP_ERROR err = report.Parse(std::move(msg));
-        ReturnErrorOnFailure(err);
+        ReturnErrorOnFailure(report.Parse(std::move(msg)));
         VerifyOrReturnError(report.GetProtocolId() == Protocols::SecureChannel::Id, CHIP_ERROR_INVALID_ARGUMENT);
 
         if (report.GetGeneralCode() == Protocols::SecureChannel::GeneralStatusCode::kSuccess &&
             report.GetProtocolCode() == Protocols::SecureChannel::kProtocolCodeSuccess && successExpected)
         {
             OnSuccessStatusReport();
+            return CHIP_NO_ERROR;
         }
-        else
-        {
-            err = OnFailureStatusReport(report.GetGeneralCode(), report.GetProtocolCode());
 
-            if (report.GetGeneralCode() == Protocols::SecureChannel::GeneralStatusCode::kBusy &&
-                report.GetProtocolCode() == Protocols::SecureChannel::kProtocolCodeBusy)
+        if (report.GetGeneralCode() == Protocols::SecureChannel::GeneralStatusCode::kBusy &&
+            report.GetProtocolCode() == Protocols::SecureChannel::kProtocolCodeBusy)
+        {
+            if (!report.GetProtocolData().IsNull())
             {
-                if (!report.GetProtocolData().IsNull())
+                Encoding::LittleEndian::Reader reader(report.GetProtocolData()->Start(), report.GetProtocolData()->DataLength());
+
+                uint16_t minimumWaitTime = 0;
+                CHIP_ERROR waitTimeErr   = reader.Read16(&minimumWaitTime).StatusCode();
+                if (waitTimeErr != CHIP_NO_ERROR)
```

Comment:
Shouldn't this read error be reported to the caller from here too?

Reply:
In practice it does not matter much, because the result will be a failure no matter what.  The question is which is the more useful failure: "got a BUSY message" or "failed to read something"?  I think the former is more useful...

Reply:
Yes, that is right. I hadn't looked inside the OnFailureStatusReport(..) function before and had thought it to be doing something else when it actually is just returning a chip error code. In that case returning from there makes sense.